### PR TITLE
R4R:modify proposal votingpower code

### PR DESCRIPTION
**Pull request by wenwu-bianjie**
https://github.com/irisnet/explorer/issues/1004

### CODE REVIEW

File: backend/orm/document/validator.go [hunk 1]
```diff
@@ -315,6 +315,26 @@ func (_ Validator) GetBondedValidators() ([]Validator, error) {
...
+	)
+
+	selector := bson.M{
+		ValidatorFieldVotingPower:     "1",
+		ValidatorFieldOperatorAddress: "",
+		ValidatorFieldDelegatorShares: "",
+		ValidatorFieldTokens:          "",
+	}
+	condition := bson.M{
+		ValidatorFieldStatus: ValidatorStatusValBonded,
```
Comment (weichang-bianjie) [inline backend/orm/document/validator.go]: bonded状态且非jailed状态

File: backend/service/service_proposal.go [hunk 2]
```diff
@@ -223,18 +378,25 @@ func (service *ProposalService) QueryList(page, size int, istotal bool) (resp vo
 		}
 
 		if propo.Status == document.ProposalStatusVoting {
+			curValidators := make(map[string]*vo.ValidatorGovInfo, len(propo.Votes))
+			curDelegators := make(map[string]*vo.DelegatorGovInfo, len(propo.Votes))
 
 			for _, v := range propo.Votes {
-				var voterVotingPower int64
-				voterInfo := addrAsMultiTypeMap[v.Voter]
+				getDelegationsByVoter(curValidators, curDelegators, v, allBondedValidators)
```
Comment (kaifei-bianjie) [inline backend/service/service_proposal.go]: set value of `curValidators`  before loop `propo.Votes`

File: backend/service/service_proposal.go [hunk 3]
```diff
@@ -133,45 +171,164 @@ func (service *ProposalService) QueryDepositAndVotingProposalList() []vo.Proposa
...
-		if propo.Status == document.ProposalStatusVoting {
-			passThreshold, vetoThreshold, participation, err := lcd.GetPassVetoThresholdAndParticipationMinDeposit(propo.Type)
+		tmp.Level = l
+
+		proposals = append(proposals, tmp)
+	}
+	return proposals
+}
+
+func getDelegationsByVoter(curValidators map[string]*vo.ValidatorGovInfo, curDelegators map[string]*vo.DelegatorGovInfo, vote document.PVote, allBondedValidators []document.Validator) float64 {
```
Comment (kaifei-bianjie) [inline backend/service/service_proposal.go]: there is no need to return `totalVotingPower`

File: backend/service/service_proposal.go [hunk 4]
```diff
@@ -133,45 +171,164 @@ func (service *ProposalService) QueryDepositAndVotingProposalList() []vo.Proposa
...
-			l.GovParam = vo.GovParam{
-				PassThreshold: passThreshold,
-				VetoThreshold: vetoThreshold,
-				Participation: participation,
+		}
+	}
+	return 0
+}
+
+func computedValidatorsPower(curValidators map[string]*vo.ValidatorGovInfo, curDelegators map[string]*vo.DelegatorGovInfo, total float64) float64 {
```
Comment (kaifei-bianjie) [inline backend/service/service_proposal.go]: there is no need to return `totalVotingPower`

File: backend/service/service_proposal.go [hunk 5]
```diff
@@ -124,7 +161,8 @@ func (service *ProposalService) QueryDepositAndVotingProposalList() []vo.Proposa
 			Type:             propo.Type,
 			Status:           propo.Status,
 			Votes:            tmpVoteArr,
-			TotalVotingPower: totalVotingPower,
+			TotalVotingPower: systemVotingPower,
+			TotalVotedPower:  totalVotedPower,
```
Comment (kaifei-bianjie) [inline backend/service/service_proposal.go]: remove field `totalVotedPower `

File: backend/service/service_proposal.go [hunk 6]
```diff
@@ -133,45 +171,164 @@ func (service *ProposalService) QueryDepositAndVotingProposalList() []vo.Proposa
...
+		logger.Error("query GetDepositProposalInitAmount", logger.String("err", err.Error()), logger.Any("depositProposalIdArr", depositProposalIdArr))
 	}
 
+	for _, propo := range proposalStatusDepositData {
+		tmp := vo.ProposalNewStyle{
+			ProposalId:       propo.ProposalId,
+			Title:            propo.Title,
+			Type:             propo.Type,
+			Status:           propo.Status,
+			TotalVotingPower: systemVotingPower,
```
Comment (kaifei-bianjie) [inline backend/service/service_proposal.go]: `TotalVotingPower` is needed?

File: backend/service/service_proposal.go [hunk 7]
```diff
@@ -86,30 +122,31 @@ func (service *ProposalService) QueryDepositAndVotingProposalList() []vo.Proposa
...
-	proposals := make([]vo.ProposalNewStyle, 0, len(data))
-	for _, propo := range data {
+	for _, propo := range proposalStatusVotingData {
 		tmpVoteArr := make([]vo.VoteWithVoterInfo, 0, len(propo.Votes))
 
+		curValidators := make(map[string]*vo.ValidatorGovInfo, len(propo.Votes))
+		curDelegators := make(map[string]*vo.DelegatorGovInfo, len(propo.Votes))
+		var totalVotedPower float64
+
 		for _, v := range propo.Votes {
```
Comment (kaifei-bianjie) [inline backend/service/service_proposal.go]: set value of curValidators before loop propo.Votes

File: backend/service/service_proposal.go (hunk 2 above)
Comment (wenwu-bianjie) [reply backend/service/service_proposal.go]: ok

File: backend/service/service_proposal.go (hunk 4 above)
Comment (wenwu-bianjie) [reply backend/service/service_proposal.go]: deleted it

File: backend/service/service_proposal.go (hunk 6 above)
Comment (wenwu-bianjie) [reply backend/service/service_proposal.go]: no

**Pull request merged**